### PR TITLE
[ICDS Dashboard] Navigation menu for mobile

### PR DESCRIPTION
**Pull request by saiprakash-v**
##### Trello card
https://trello.com/c/mbNYO9oF

##### SUMMARY
Updated navigation menu for mobile dashboard for links to work

### CODE REVIEW

**Comment by czue** [comment]
Do you know why all these merge commits are in here?

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html [hunk 1]
```diff
@@ -1,43 +1,189 @@
...
-        <img src="{% static "icds_reports/mobile/images/program-summary.png" %}" class="mr-3">
-        <p class="navigation-font mb-0">Program Summary</p>
-    </a>
-    <div class="menu-options">
-        <img src="{% static "icds_reports/mobile/images/mother.png" %}" class="mr-3">
-        <p class="navigation-font mb-0">Maternal and Child Nutrition</p>
+    <img src="{% static 'icds_reports/mobile/images/white-close-icon.png' %}" class="close-icon" ng-click="closeMenu()">
+    <p class="navigation-font pl-3 pr-3 mb-0" ng-bind="userFullName" ng-if="userFullName"></p>
+    <p class="navigation-font pl-3 pr-3 mb-4" ng-bind="userUsername"></p>
+    <div style="overflow: auto; height: calc(100vh - 100px)">
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: is it required that this be an inline style declaration?

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html [hunk 2]
```diff
@@ -1,43 +1,189 @@
...
+        <a class="menu-options" href="{$ goToStep('', $location.search()) $}" ng-click="closeMenu()">
+            <img src="{% static 'icds_reports/mobile/images/program-summary.png' %}" class="mr-3">
+            <p class="navigation-font mb-0">Program Summary</p>
+        </a>
+        <div class="menu-options" ng-click="expandedSectionId='nutrition'">
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: how much of this is shared with the web navigation? this is probably out of scope for this PR, but is it plausible to imagine making `navigation-section.directive` and `navigation-item.directive` and then reusing the same data structure to create both sets of menus at some point in the future?

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html [hunk 3]
```diff
@@ -1,43 +1,189 @@
...
+                </div>
+                <div class="navigation-font pl-3 pr-3 mb-0"
+                     ng-class="{'active-mobile-link': $location.path().indexOf('/awc_infrastructure/infants_weight_scale') !== -1}">
+                    <a href="{$ goToStep('awc_infrastructure/infants_weight_scale', $location.search()) $}" class="sub-links" ng-click="closeMenu()">
+                      AWCs Reported Weighing Scale: Infants</a>
+                </div>
+                <div class="navigation-font pl-3 pr-3 mb-0"
+                     ng-class="{'active-mobile-link': $location.path().indexOf('/awc_infrastructure/adult_weight_scale') !== -1}">
+                    <a href="{$ goToStep('awc_infrastructure/adult_weight_scale', $location.search()) $}" class="sub-links" ng-click="closeMenu()">
+                      AWCs Reported mother and child weighing scale</a>
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: I think some of these titles need to be reconciled with the latest changes on web. See, e.g.
https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/templates/icds_reports/icds_app/navigation.directive.html#L124

**Review by czue — CHANGES_REQUESTED** [review]
Looks good. For now, can you recheck all the page titles against web navigation (it changed in the last day)? Then I think this should be good to go as a first pass - though would be nice if we could get some more reuse since there's so much in common and it will be annoying to keep both of these in sync.

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html (hunk 1 above)
Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: I will update it

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html (hunk 3 above)
Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: Sure. Will update it

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html (hunk 2 above)
Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: I'm thinking of making a constant (array of objects with menu names and corresponding routes) in one of the shared utils and iterate over it in both web and mobile navigation.
Let me know your take on this.

Comment (czue) [reply custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: That sounds great, but can we do it in a follow up so we can demo this in the meantime?

**Review by czue — APPROVED** [review]

**Pull request merged**